### PR TITLE
Add packer validate github check workflow

### PR DESCRIPTION
**Pull request by peterzhuamazon**
### Description
Add packer validate github check workflow

### Issues Resolved
Closes https://github.com/opensearch-project/opensearch-ci/issues/438

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .github/workflows/packer.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Setup `packer`
+        uses: hashicorp/setup-packer@main
+        id: setup
+        with:
+          version: ${{ env.PACKER_VERSION }}
+
+      - name: Run `packer validate`
```
Comment (gaiksaya) [inline .github/workflows/packer.yml:25]:
```suggestion
      - name: Validate packer
```

File: .github/workflows/packer.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        id: validate
+        run: |
+            cd packer/
```
Comment (gaiksaya) [inline .github/workflows/packer.yml:28]: You can add working directory at the workflow level as well.
https://github.com/opensearch-project/opensearch-devops/blob/main/.github/workflows/nightly-playground-test.yml#L13

```suggestion
             working-directory: packer
```

File: .github/workflows/packer.yml (hunk 1 above)
Comment (peterzhuamazon) [reply .github/workflows/packer.yml:25]: Updated. Thanks.

File: .github/workflows/packer.yml (hunk 2 above)
Comment (peterzhuamazon) [reply .github/workflows/packer.yml:28]: Updated. Thanks.

File: packer/jenkins-agent-al2-arm64.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "variables": {
-    "name-base":"Jenkins-Agent-AL2-ARM64-JDK21",
+    "name-base":"Jenkins-Agent-AL2-ARM64",
```
Comment (gaiksaya) [inline packer/jenkins-agent-al2-arm64.json:3]: What were name-base used for? Just want to make sure we are not breaking any existing funtionality

File: .github/workflows/packer.yml [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+        id: setup
+        with:
+          version: ${{ env.PACKER_VERSION }}
+
+      - name: Validate Packer Templates
+        id: validate
+        working-directory: packer
+        run: |
+            for template in `ls | grep ".json"`; do
+                echo $template
```
Comment (gaiksaya) [inline .github/workflows/packer.yml:30]: Do we need to echo the entire template or was this for debugging purpose?

Comment (peterzhuamazon) [reply .github/workflows/packer.yml:30]: It is just echoing the name of the template, not the content.

Comment (peterzhuamazon) [reply .github/workflows/packer.yml:30]:
```
jenkins-agent-al2-arm64.json
The configuration is valid.
jenkins-agent-al2-x64.json
The configuration is valid.
jenkins-agent-al2023-arm64.json
The configuration is valid.
jenkins-agent-al2023-x64.json
The configuration is valid.
jenkins-agent-macos12-x64.json
The configuration is valid.
jenkins-agent-ubuntu2004-x64.json
The configuration is valid.
jenkins-agent-win2019-x64-gradle-check.json
The configuration is valid.
jenkins-agent-win2019-x64.json
The configuration is valid.

```

File: packer/jenkins-agent-al2-arm64.json (hunk 3 above)
Comment (peterzhuamazon) [reply packer/jenkins-agent-al2-arm64.json:3]: We dont need to add those into the AMI name itself.
As these are just templates to run scripts.
The scripts itself already contain the JDK version it needs to install.

Comment (gaiksaya) [reply packer/jenkins-agent-al2-arm64.json:3]: I believe we are using AMI ID in our configs so this change wont affect us or any community user using jenkins specs right?

Comment (peterzhuamazon) [reply packer/jenkins-agent-al2-arm64.json:3]: Yes, It wont.

Comment (peterzhuamazon) [reply packer/jenkins-agent-al2-arm64.json:3]: The only thing matters is the AMI id here.

**Pull request merged**